torch/distributed/_composable/fsdp/_fsdp_state.py

```diff
@@ -106,10 +188,13 @@ def _lazy_init(self) -> None:
             self._state_ctx.all_states.append(state)
             if state._fsdp_param_group:
                 state._fsdp_param_group.lazy_init()
-        if self._fsdp_param_group:
-            # For the root, do not reshard after forward since for training,
-            # the parameters would be freed and all-gathered immediately
-            self._fsdp_param_group.post_forward_mesh_info = None
+        # NOTE(yf225): I believe this causes issue during compile
+        # (post_forward hook thinks that `post_forward_mesh_info` is None thus not doing resharding and not doing .resize_(0). Is it expected eager behavior?)
```

Comment:
This is expected behavior. We choose to not call `resize_(0)` on the root FSDP module's parameters after forward since they would be immediately `resize_()`-d back and all-gathered.